src/js/containers/dashboard/LastDateModifiedContainer.jsx

```diff
@@ -0,0 +1,97 @@
+/**
+ * LastDateModifiedContainer.jsx
+ * Created by Kwadwo Opoku-Debrah 9/30/18
+ */
+
+import React, { PropTypes } from 'react';
+import { bindActionCreators } from 'redux';
+import { connect } from 'react-redux';
+
+import _ from 'lodash';
+
+import * as lastDateModifiedActions from '../../redux/actions/lastDateModifiedActions';
+import * as lastDateModifiedHelper from '../../helpers/lastDateModifiedHelper';
+
+import CalendarRangeDatePicker from '../../components/SharedComponents/CalendarRangeDatePicker';
+
+const propTypes = {
+  setLastDateModifiedList: PropTypes.func,
+  lastDateModifiedList: PropTypes.object,
+  detached: PropTypes.bool,
+  selectedFilters: PropTypes.object,
+  type: PropTypes.string,
+  table: PropTypes.string,
+  placeholder: PropTypes.string,
+  onSelect: PropTypes.func
+};
+
+const defaultProps = {
+  setLastDateModifiedList: () => {},
+  lastDateModifiedList: {},
+  detached: true,
+  selectedFilters: [],
+  table: '',
+  type: '',
+  placeholder: '',
+  onSelect: () => {}
+};
+
+class LastDateModifiedContainer extends React.Component {
+  componentDidMount() {
+    this.loadData();
+  }
+
+  loadData() {
+    if (_.isEmpty(this.props.lastDateModifiedList.lastDateModified)) {
+      // we need to populate the list
+      if (this.props.detached) {
+        lastDateModifiedHelper.fetchLastDateModified()
+          .then((data) => {
+            this.props.setLastDateModifiedList(data);
```

Comment:
If all you need is the min and max modified date, it would be a better pattern to pass `data` to a parse function and store only what you need in the component state, rather than the entire API response in Redux.